frontend/src/pages/projects/screens/detail/data-connections/ManageDataConnectionModal.tsx

```diff
@@ -76,52 +76,66 @@ const ManageDataConnectionModal: React.FC<ManageDataConnectionModalProps> = ({
       existingData?.data.metadata.name,
     );
 
-    const promiseActions: Promise<K8sResourceCommon>[] = [];
     const secretName = assembledSecret.metadata.name;
 
-    if (existingData) {
-      if (AWSDataChanged) {
-        promiseActions.push(replaceSecret(assembledSecret));
+    const runPromiseActions = async (dryRun: boolean) => {
+      const promiseActions: Promise<K8sResourceCommon>[] = [];
+
+      if (existingData) {
+        if (AWSDataChanged) {
+          promiseActions.push(replaceSecret(assembledSecret, { dryRun: dryRun }));
+        }
+
+        const notebooksToDisconnect = allAvailableNotebooks.filter((notebook) =>
+          removedConnections.includes(notebook.metadata.name),
+        );
+
+        promiseActions.push(
+          ...notebooksToDisconnect.map((notebook) =>
+            replaceNotebookSecret(
+              notebook.metadata.name,
+              projectName,
+              getSecretsFromList(notebook).filter(({ secretRef: { name } }) => name !== secretName),
+              { dryRun: dryRun },
+            ),
+          ),
+        );
+      } else {
+        promiseActions.push(createSecret(assembledSecret, { dryRun: dryRun }));
       }
 
-      const notebooksToDisconnect = allAvailableNotebooks.filter((notebook) =>
-        removedConnections.includes(notebook.metadata.name),
+      const notebooksToConnect = allAvailableNotebooks.filter((notebook) =>
+        addedConnections.includes(notebook.metadata.name),
       );
+
       promiseActions.push(
-        ...notebooksToDisconnect.map((notebook) =>
-          replaceNotebookSecret(
+        ...notebooksToConnect.map((notebook) =>
+          attachNotebookSecret(
             notebook.metadata.name,
             projectName,
-            getSecretsFromList(notebook).filter(({ secretRef: { name } }) => name !== secretName),
+            secretName,
+            hasEnvFrom(notebook),
+            { dryRun: dryRun },
           ),
         ),
       );
-    } else {
-      promiseActions.push(createSecret(assembledSecret));
-    }
-
-    const notebooksToConnect = allAvailableNotebooks.filter((notebook) =>
-      addedConnections.includes(notebook.metadata.name),
-    );
 
-    promiseActions.push(
-      ...notebooksToConnect.map((notebook) =>
-        attachNotebookSecret(notebook.metadata.name, projectName, secretName, hasEnvFrom(notebook)),
-      ),
-    );
+      if (promiseActions.length > 0) {
+        setErrorMessage('');
+        setIsProgress(true);
+        return await Promise.all(promiseActions)
+          .then(() => Promise.resolve())
+          .catch((e) => {
+            setErrorMessage(e.message || 'An unknown error occurred');
+            setIsProgress(false);
+            return Promise.reject();
```

Comment:
```suggestion
            throw e;
```
I imagine you want to go to the catch of the caller not the then... does returning a reject promise kick the caller's catch or the then? I guess I have not ever seen this code in practice now that I think of it haha.

Reply:
I think its falls into the catch... It is the same as throwing an error except I felt an error wasnt right in this situation. For example. There should not be an error if there were no actions. It also is more consistent with the Promise.resolve()

Reply:
Interesting -- I don't typically use `Promise.resolve` or `Promise.reject` outside of the root of an async method -- looked odd to me, but I suppose it would be fairly simple of a code block to just invoke the catch statement with a throw internally anyways.

Reply:
would you rather have the async function propagate the error into the outer catch block instead of inside the function